src/EntityFrameworkCoreMock.Moq/DbContextMock.cs

```diff
@@ -62,12 +62,17 @@ public void Reset()
             Setup(x => x.SaveChangesAsync(It.IsAny<bool>(), It.IsAny<CancellationToken>())).ReturnsAsync(SaveChanges);
             Setup(x => x.SaveChangesAsync(It.IsAny<CancellationToken>())).ReturnsAsync(SaveChanges);
 
-            var mockDbFacade = new Mock<DatabaseFacade>(Object);
-            var mockTransaction = new Mock<IDbContextTransaction>();
-            mockDbFacade.Setup(x => x.BeginTransaction()).Returns(mockTransaction.Object);
-            mockDbFacade.Setup(x => x.BeginTransactionAsync(It.IsAny<CancellationToken>()))
-                .Returns(Task.FromResult(mockTransaction.Object));
-            Setup(x => x.Database).Returns(mockDbFacade.Object);
+            var lazyMockDbFacade = new Lazy<Mock<DatabaseFacade>>(() =>
+            {
+                var mockDbFacade = new Mock<DatabaseFacade>(Object);
+                var mockTransaction = new Mock<IDbContextTransaction>();
+                mockDbFacade.Setup(x => x.BeginTransaction()).Returns(mockTransaction.Object);
+                mockDbFacade.Setup(x => x.BeginTransactionAsync(It.IsAny<CancellationToken>()))
+                    .Returns(Task.FromResult(mockTransaction.Object));
```

Comment:
Not sure why Returns was used originally here
```suggestion
                    .ReturnsAsync(mockTransaction.Object);
```

Reply:
Good suggestion, thanks.